backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/ui/web/order/BOMExploderCommand.java

```diff
@@ -130,17 +139,59 @@ public BOMExploderCommand(@Nullable final BOMUse bomToUse,
 			final OrderLineCandidate lineCandidate = initialCandidate.toBuilder()
 					.productId(bomLineProductId)
 					.attributes(attributes)
-					.qty(bomLineQty)
+					.qty(Quantitys.create(bomLineQty, bomUomId))
 					.compensationGroupId(compensationGroupId)
-					.uomId(bomUomId)
 					.explodedFromBOMLineId(bomLineId)
 					.build();
 			result.addAll(this.toBuilder()
 					.initialCandidate(lineCandidate)
 					.build()
-					.execute());  // recurse
+					.execute0());  // recurse
 		}
 
 		return result;
 	}
+
+	private List<OrderLineCandidate> reduceCandidates(final List<OrderLineCandidate> result)
+	{
+		final ArrayList<OrderLineCandidate> reduced = new ArrayList<>();
+		for (final OrderLineCandidate candidate : result)
+		{
+			if (candidate.getQty().signum() == 0)
+			{
+				continue;
+			}
+			if (candidate.getQty().signum() < 0)
+			{
+				throw new AdempiereException("Invalid qty: " + candidate.getQty());
+			}
+			if (candidate.getQty().toBigDecimal().compareTo(BigDecimal.ZERO) == 0)
+			{
+				continue;
+			}
+
+			final Optional<OrderLineCandidate> orderLineCandidate = reduced.stream()
+					.filter(cand -> cand.getProductId().equals(candidate.getProductId()))
+					.findFirst();
+			if (orderLineCandidate.isPresent())
+			{
+				final OrderLineCandidate olCandidate = orderLineCandidate.get();
```

Comment:
about:
* Optional<OrderLineCandidate> orderLineCandidate
* OrderLineCandidate olCandidate = orderLineCandidate.get();

I find it a bit misleading/hard to read because we have similar but not suggestive names to distinguish the same information in different formats.

I would prefer on of the following approaches:

Approach 1. Avoid optional vars (kind of my preferate)
i.e. 
OrderLineCandidate orderLineCandidate = reduced.stream()... .findFirst().orElse(null);
...
if(orderLineCandidate != null) ....

Approach 2. Better var names
OrderLineCandidate optionalOrderLineCandidate = reduced.stream()... .findFirst();
....

Reply:
Changed to use Approach 1.